### ISSUE TITLE
Refactor user tier/token config to clean structure

### ISSUE THREAD

**Issue by PatrickR315516** [apple/live-caller-id-lookup-example#52]
Since the ‘users’ object does not correspond to a JSON structure in best practices, I suggest that it be transformed into a better structure.

Replacing the `users` element to an array of objects with `tier` and `tokens` attributes in [`service-config.json`](./service-config.json):
```json
{
...
  "users": [
    {
      "tier": "tier1",
      "tokens": ["AAAA"]
    },
    {
      "tier": "tier2",
      "tokens": ["BBBB", "CCCC"]
    }
  ],
...
}
```

Replacing the `ServerConfiguration` in [`PIRService/server.swift`](./Sources/PIRService/server.swift)
```swift
struct ServerConfiguration: Codable {
    struct Usecase: Codable {
        let name: String
        let fileStem: String
        let shardCount: Int
    }
    
    struct User: Codable {
        let tier: UserTier
        let tokens: [String]
    }

    let users: [User]
    let usecases: [Usecase]
}
```

And modify the `run` method in [the same file](./Sources/PIRService/server.swift) to: 
```swift
...
let config = try JSONDecoder().decode(ServerConfiguration.self, from: configData)

if !config.users.isEmpty {
    let authenticator = UserAuthenticator()
    for user in config.users {
        for token in user.tokens {
            await authenticator.add(token: token, tier: user.tier)
        }
    }
    privacyPassState = try .init(userAuthenticator: authenticator)
}

...
```

**Comment by karulont** [comment]
Thank you! I think this is a good change. You are welcome to open a pull request.

**Comment by karulont** [comment]
Fixed in #54
